### PR TITLE
Make some renames in js/build.ts

### PR DESCRIPTION
**Pull request by ry**
See discussion in #1909
cc @kitsonk @kevinkassimo @zekth

### CODE REVIEW

File: js/build.ts [hunk 1]
```diff
@@ -4,30 +4,26 @@
...
-  mac = "mac",
-  win = "win",
-  linux = "linux"
-}
+export type OperatingSystem = "mac" | "win" | "linux";
 
 // 'build' is injected by rollup.config.js at compile time.
 export const build: BuildInfo = {
-  /* eslint-disable @typescript-eslint/no-explicit-any */
   // These string will be replaced by rollup
```
Comment (piscisaureus) [inline js/build.ts]: If you use disable-next-line then you don't need to re-enable below.

Comment (ry) [reply js/build.ts]: oops - fixed.

File: js/build.ts [hunk 2]
```diff
@@ -4,30 +4,26 @@
 /** Build related information */
 export interface BuildInfo {
   /** The operating system CPU architecture. */
-  arch: "x64";
+  arch: "x64" | "arm64";
```
Comment (kitsonk) [inline js/build.ts]: We should probably create a type for arch as well?

```ts
export type Architecture = "x64" | "arm64";
```

Comment (ry) [reply js/build.ts]: done

**Pull request merged**